### PR TITLE
EDM-555: add resource monitor API validation

### PR DESCRIPTION
**Pull request by hexfusion**
This PR adds `ResourceMonitor` validation rules and fixes a bug where we allow defining time duration in days which is not supported. Instead of adding that support, it makes sense to allow the largest interval to be in hours as days are not percise.

Ref: golang/go#17767

### CODE REVIEW

File: internal/util/validation/validation.go [hunk 1]
```diff
@@ -282,3 +282,18 @@ func asErrors(errs field.ErrorList) []error {
 	}
 	return agg.Errors()
 }
+
+// CheckUnique checks for duplicate values in a slice of items.
+func CheckUnique[T any](slice []T, fieldFn func(T) string) error {
```
Comment (avishayt) [inline internal/util/validation/validation.go:287]: Where is this used?

**Review by avishayt — COMMENTED** [review]
Overall looks good.  What was the issue with days?

File: internal/util/validation/validation.go (hunk 1 above)
Comment (hexfusion) [reply internal/util/validation/validation.go:287]: Meant to remove

**Comment by hexfusion** [comment]
> Overall looks good.  What was the issue with days?

Days are not constant length in every location.

https://github.com/golang/go/issues/17767

**Comment by avishayt** [comment]
> > Overall looks good.  What was the issue with days?
> 
> Days are not constant length in every location.
> 
> [golang/go#17767](https://github.com/golang/go/issues/17767)

Interesting - thanks!

File: internal/util/validation/validation.go (hunk 1 above)
Comment (hexfusion) [reply internal/util/validation/validation.go]: Done

**Review by avishayt — APPROVED** [review]

**Pull request merged**